meson.build

```
@@ -0,0 +1,168 @@
+    '--vapidir', vapi_dir,
+    '--enable-experimental',
+    '--disable-warnings',
+    '--target-glib=2.32',
+    '--thread',
+    '-g',
+    '--save-temps'
+], language: 'vala')
+
+
+src = [
+    gresource,
+    'src' / 'config.vala',
+    'src' / 'Controller.vala',
+    'src' / 'Library.vala',
+    'src' / 'MainWindow.vala',
+    'src' / 'Objects' / 'DirectoryEntry.vala',
+    'src' / 'Objects' / 'Episode.vala',
+    'src' / 'Objects' / 'Podcast.vala',
+    'src' / 'Services' / 'ImageCache.vala',
+    'src' / 'Utils' / 'FeedParser.vala',
+    'src' / 'Utils' / 'gpodderClient.vala',
+    'src' / 'Utils' / 'iTunesProvider.vala',
+    'src' / 'Utils' / 'MPRIS.vala',
+    'src' / 'Utils' / 'PasswordManager.vala',
+    'src' / 'Utils' / 'Player.vala',
+    'src' / 'Utils' / 'SoupClient.vala',
+    'src' / 'Utils' / 'Utils.vala',
+    'src' / 'Utils' / 'XmlUtils.vala',
+    'src' / 'Vocal.vala',
+    'src' / 'VocalSettings.vala',
+    'src' / 'Widgets' / 'AddFeedDialog.vala',
+    'src' / 'Widgets' / 'ArtworkPopover.vala',
+    'src' / 'Widgets' / 'CoverArt.vala',
+    'src' / 'Widgets' / 'DirectoryArt.vala',
+    'src' / 'Widgets' / 'DirectoryView.vala',
+    'src' / 'Widgets' / 'DownloadDetailBox.vala',
+    'src' / 'Widgets' / 'DownloadsPopover.vala',
+    'src' / 'Widgets' / 'EpisodeDetailBox.vala',
+    'src' / 'Widgets' / 'InternetArchiveUploadDialog.vala',
+    'src' / 'Widgets' / 'NewEpisodesView.vala',
+    'src' / 'Widgets' / 'PlaybackBox.vala',
+    'src' / 'Widgets' / 'PodcastView.vala',
+    'src' / 'Widgets' / 'QueueBox.vala',
+    'src' / 'Widgets' / 'QueueRow.vala',
+    'src' / 'Widgets' / 'SearchResultBox.vala',
+    'src' / 'Widgets' / 'SearchResultsView.vala',
+    'src' / 'Widgets' / 'SettingsDialog.vala',
+    'src' / 'Widgets' / 'Shownotes.vala',
+    'src' / 'Widgets' / 'SyncDialog.vala',
+    'src' / 'Widgets' / 'Toolbar.vala',
+    'src' / 'Widgets' / 'VideoControls.vala',
+]
+
+dependencies = [
+    dependency('clutter-gst-3.0'),
+    dependency('clutter-gtk-1.0'),
+    dependency('gee-0.8'),
+    dependency('granite'),
+    dependency('gstreamer-1.0'),
+    dependency('gstreamer-pbutils-1.0'),
+    dependency('gtk+-3.0'),
+    dependency('glib-2.0'),
+    dependency('gobject-2.0'),
+    dependency('json-glib-1.0'),
+    dependency('libsecret-1'),
+    dependency('libsoup-2.4'),
+    dependency('libxml-2.0'),
+    dependency('sqlite3'),
+    dependency('webkit2gtk-4.0'),
+    # meson.get_compiler('vala').find_library('webkit2gtk-3.0', dirs: vapi_dir)
+]
+
+
+ss = ssmod.source_set()
+ss.add(src)
+ss.add(dependencies)
+ss.add(
+    when: dependency('unity'), 
+    if_true: dependency('unity'),
+)
+ss.add(
+    when: dependency('libnotify'), 
+    if_true: dependency('libnotify'),
+)
+config = configuration_data()
+ssconfig = ss.apply(config)
+
+i18n.merge_file(
+    input: 'data' / 'com.github.needleandthread.vocal.desktop.in',
+    output: meson.project_name() + '.desktop',
+    po_dir: meson.source_root() / 'po',
```

source_root() is unnecessary here, since that is your current directory anyway.

```suggestion
    po_dir: 'po',
```